Automattic/jetpack
# Sync: Updated HPOS Orders Module to allow shop_subscription type

Fixes https://github.com/Automattic/vulcan/issues/462

## Proposed changes:

* Added `shop_subscription` type to `get_order_types_to_sync`

### Other information:


## Jetpack product discussion


p9dueE-8ec-p2#comment-10532
## Does this pull request change what data or activity we track or use?



## Testing instructions:





Code review should be sufficient here.
To really test, and since I'm not very familiar with Woo Subscriptions I suggest to directly tweak the order type directly in the DB:
Start with a Jetpack connected site with WooCommerce, HPOS enabled, and a few orders. Follow the pre-requisites in D154724-code to run a checksum.
- Then, before applying this PR just change one order directly in the DB in your testing site so that its `type` is `shop_subscription`.
- Try running a checksum fix again for `wc_orders`. The order id will show now as missing in remote. The reason is that for checksum purposes we don't include orders with `shop_subscription` yet. The checksum fix will actually remove the order from the cache site, since remote is the source of truth. If run again, no differences or missing orders will be found.
- Apply the PR
- Run once more the checksum fix for `wc_orders`. Now it will show that it is missing on wpcom. It makes sense since we removed it from the cached site. The fix will add it to wpcom. 
If you want to keep your test site with the initial value. Just revert the change in the DB and run a last checksum. It will mark that there is a difference since it exists in both sites but with different value and will copy the one from remote.

**Note:** Even though most of the steps above work, a fatal can occur when trying to retrieve the order with `shop_subscription` type if the WooCommerce Subscriptions extension is not present on the site. Context: p1724334770338649-slack-C055WHLA98D
